examples/nlp/multi_label_classification.py

```diff
@@ -0,0 +1,431 @@
+    arxiv_data_filtered,
+    test_size=test_split,
+    stratify=arxiv_data_filtered["terms"].values,
+)
+
+# Splitting the test set further into validation
+# and new test sets.
+val_df = test_df.sample(frac=0.5)
+test_df.drop(val_df.index, inplace=True)
+
+print(f"Number of rows in training set: {len(train_df)}")
+print(f"Number of rows in validation set: {len(val_df)}")
+print(f"Number of rows in test set: {len(test_df)}")
+
+"""
+## Multi-label binarization
+
+Now we preprocess our labels using the
+[`StringLookup`](https://keras.io/api/layers/preprocessing_layers/categorical/string_lookup)
+layer. 
+"""
+
+terms = tf.ragged.constant(train_df["terms"].values)
+lookup = tf.keras.layers.StringLookup(output_mode="multi_hot")
+lookup.adapt(terms)
+vocab = lookup.get_vocabulary()
+
+
+def invert_multi_hot(encoded_labels):
+    """Reverse a single multi-hot encoded label to a tuple of vocab terms."""
+    hot_indices = np.argwhere(encoded_labels == 1.0)[..., 0]
+    return np.take(vocab, hot_indices)
+
+
+print("Vocabulary:\n")
+print(vocab)
+
+
+"""
+Here we are separating the individual unique classes available from the label
+pool and then using this information to represent a given label set with 0's and 1's.
+Below is an example.
+"""
+
+sample_label = train_df["terms"].iloc[0]
+print(f"Original label: {sample_label}")
+
+label_binarized = lookup([sample_label])
+print(f"Label-binarized representation: {label_binarized}")
+
+"""
+## Data preprocessing and `tf.data.Dataset` objects
+
+We first get percentile estimates of the sequence lengths. The purpose will be clear in a
+moment.
+"""
+
+train_df["summaries"].apply(lambda x: len(x.split(" "))).describe()
+
+"""
+Notice that 50% of the abstracts have a length of 154 (you may get a different number
+based on the split). So, any number near that is a good enough approximate for the
+maximum sequence length.
+
+Now, we write utilities to prepare our datasets that would go straight to the text
+classifier model.
+"""
+
+max_seqlen = 150
+batch_size = 128
+padding_token = "<pad>"
+auto = tf.data.AUTOTUNE
+
+
+def unify_text_length(text, label):
+    # Split the given abstract and calculate its length.
+    word_splits = tf.strings.split(text, sep=" ")
+    sequence_length = tf.shape(word_splits)[0]
+
+    # Calculate the padding amount.
+    padding_amount = max_seqlen - sequence_length
+
+    # Check if we need to pad or truncate.
+    if padding_amount > 0:
+        unified_text = tf.pad([text], [[0, padding_amount]], constant_values="<pad>")
+        unified_text = tf.strings.reduce_join(unified_text, separator="")
+    else:
+        unified_text = tf.strings.reduce_join(word_splits[:max_seqlen], separator=" ")
+
+    # The expansion is needed for subsequent vectorization.
+    return tf.expand_dims(unified_text, -1), label
+
+
+def make_dataset(dataframe, is_train=True):
+    labels = tf.ragged.constant(dataframe["terms"].values)
+    label_binarized = lookup(labels).numpy()
```

Comment:
Do you actually need the numpy() call here? I think from_tensor_slices should handle the tensor output directly just fine.

Reply:
Yeah, it should. But I find it a good practice to use uniform data types while calling `from_tensor_slices()`.